plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3138,28 +3138,32 @@ private Optional<SystemTable> getRawSystemTable(ConnectorSession session, Schema
         }
 
         String tableLocation = table.get().location();
-        TableSnapshot tableSnapshot = getSnapshot(new SchemaTableName(systemTableName.getSchemaName(), tableName), tableLocation, session);
-        MetadataEntry metadataEntry;
-        try {
-            metadataEntry = transactionLogAccess.getMetadataEntry(tableSnapshot, session);
```

Comment:
Moving this into `case PROPERTIES` could be a separate commit. It would be easier to understand that change isn't related to other changes (it took me a while to understand that)